src/hidimstat/conditional_permutation_importance.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.base import check_is_fitted, clone
+from sklearn.metrics import root_mean_squared_error
+
+from hidimstat.base_perturbation import BasePerturbation
+from hidimstat.conditional_sampling import ConditionalSampler
+
+
+class CPI(BasePerturbation):
+    def __init__(
+        self,
+        estimator,
+        loss: callable = root_mean_squared_error,
+        method: str = "predict",
+        n_jobs: int = 1,
+        n_permutations: int = 50,
+        imputation_model_continuous=None,
+        imputation_model_categorical=None,
+        random_state: int = None,
+        categorical_max_cardinality: int = 10,
+    ):
+        """
+        Conditional Permutation Importance (CPI) algorithm.
+        :footcite:t:`Chamma_NeurIPS2023` and for group-level see
+        :footcite:t:`Chamma_AAAI2024`.
+
+        Parameters
+        ----------
+        estimator : sklearn compatible estimator, optional
+            The estimator to use for the prediction.
+        loss : callable, default=root_mean_squared_error
+            The loss function to use when comparing the perturbed model to the full
+            model.
+        method : str, default="predict"
+            The method to use for the prediction. This determines the predictions passed
+            to the loss function. Supported methods are "predict", "predict_proba",
+            "decision_function", "transform".
+        n_jobs : int, default=1
+            The number of jobs to run in parallel. Parallelization is done over the
+            variables or groups of variables.
+        n_permutations : int, default=50
+            The number of permutations to perform. For each variable/group of variables,
+            the mean of the losses over the `n_permutations` is computed.
+        imputation_model_continuous : sklearn compatible estimator, optional
+            The model used to estimate the conditional distribution of a given
+            continuous variable/group of variables given the others.
+        imputation_model_categorical : sklearn compatible estimator, optional
+            The model used to estimate the conditional distribution of a given
+            categorical variable/group of variables given the others. Binary is
+            considered as a special case of categorical.
+        random_state : int, default=None
+            The random state to use for sampling.
+        categorical_max_cardinality : int, default=10
+            The maximum cardinality of a variable to be considered as categorical
+            when the variable type is inferred (set to "auto" or not provided).
```

Comment:
Can you add a section with the reference?

Reply:
What do you mean?

Reply:
```
    References
    ----------
    .. footbibliography::
```
When you cite a paper, it's better to add the previous section for having the footcitation.